src/frontend/mosh-server.cc

```diff
@@ -336,6 +339,26 @@ int main( int argc, char *argv[] )
 static int run_server( const char *desired_ip, const char *desired_port,
 		       const string &command_path, char *command_argv[],
 		       const int colors, bool verbose, bool with_motd ) {
+  /* get network idle timeout */
+  long network_timeout = 0;
+  char *timeout_envar = getenv( "MOSH_SERVER_NETWORK_TMOUT" );
+  if ( timeout_envar ) {
+    /* We don't care about strtol errors here, they all produce sensible return values. */
+    network_timeout = strtol( timeout_envar, NULL, 10 );
+    if ( network_timeout < 0 ) {
+      network_timeout = 0;
```

Comment:
You have a good point about the silent error.  But having a bad value mean "exit immediately" results in another problem (at least for bash and `TMOUT`):  if you pass a bad value from your `.profile`, then your shell will exit immediately next time you login, and you will have an interesting time fixing that.  It's less of an issue here, since the timeout only operates on lost network connectivity, but we might then surprise people by operating differently than `TMOUT`.  I'll think about this one.